### DISCUSSION TITLE
Local jamulus for wireless monitoring

### DISCUSSION THREAD

**Discussion by ngocdh** [jamulussoftware/jamulus#1230] [Ideas]
First of all, I would like to send a big THANK YOU to the original author and the dev team. Jamulus is so much fun, even for a beginner like me.

I've been studying Jamulus and all of its possibilities. I must say the potential is great! 

Now that we have a working Android version (thanks again to those involved). Although delay is still important for general use, it should be very possible for wireless monitoring (musicians are in the same room), because now we're in a local network. Two scenarios came to mind:

- For practice or quick jam: a computer acts as server via a local wireless network. Each musicians can use their android device to connect to the server via its IP address. Of course you'll need an adapter to plug your instrument into the phone, for me irig works well. In this scenario, we don't even need a mixer, Jamulus is our mixer. Each musician has their own mix they can control using their phone.

- Second scenario is quite similar, but for bands who want guaranty on input quality, or for recording. Now a mixer is required and is connected to the server via usb. Each musician will connect to the server using their phone, but as listener only, as their instruments are plugged into the mixer (or they can use their built-in mic for conversation when needed, because it's hard to hear people when using in-ear monitoring).  However, in this scenario, musicians no longer have access to their individual mix like in scenario 1. A work around for this is to create multiple client instances on the server, so that each input has its own fader.

I know people have been asking for this feature, and I would really love one day to see Jamulus allow multiple faders per client when using digital mixer, so that this second scenario is supported natively. 

I'm sure some of you will recognize the second scenario: yes the idea is inspired by Audio Fusion wireless monitoring system, which already lower greatly the cost. But if we can do better with Jamulus, then why not?

**Comment by ann0see** [comment]
Great to hear that! 

Concerning your feature request with multiple sliders: I understand you want it and multiple other (more professional people also would want it), but this might make Jamulus much more complicated (code wise and UI wise). Therefore, since it is an advanced feature request you could also use an advanced (hardware) mixer setup instead? 

Maybe your feature request can also be fulfilled with the single mix approach (see the branch).

**Reply by ngocdh** [reply]
Thanks for your suggestion. As I stated above, single-mix wireless monitoring is already possible with Jamulus' current state.
Just wanted to point out the potential if we have multiple sliders. 
Code-wise, yes Jamulus client will have to be able to read inputs from digital mixers, I think this is the hard part. About UI, when each input has its own slider, maybe the possibility to name the inputs should be enough?
It seems though this is not something the team plans to implement?

**Reply by ann0see** [reply]
The problem here is that something like that might be out of scope and even confuse new, inexperienced users?

You might be able to use midi controllers soon, but more than two inputs is currently out of scope of Jamulus. You can of course fork Jamulus, do some work yourself and then open a Pull request. But I doubt it's being accepted at least at the moment. 

Please also keep in mind that all the mixing work is done on the server which currently only supports two inputs and two outputs.

**Reply by ngocdh** [reply]
Thanks for your inputs @ann0see
